### PR TITLE
Trev8939/feature linked anno

### PR DESCRIPTION
**Pull request by TADraeseke**
Hi @hud10837 - This one is ready for review. Please have a look at the code files mostly. Since this is a new design, errors in the README are probably best taken in the common samples PR https://devtopia.esri.com/runtime/common-samples/pull/2103

Thanks!

### CODE REVIEW

File: kotlin/edit-features-with-feature-linked-annotation/src/main/java/com/esri/arcgisruntime/sample/editfeatureswithfeaturelinkedannotation/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    AlertDialog.Builder(this).apply {
+      setTitle("Edit feature attribute:")
+      setView(editAttributeView)
+      // populate edit texts with current attribute values
+      editAttributeView.addressNumberEditText.setText(selectedFeature.attributes["AD_ADDRESS"].toString())
+      editAttributeView.streetEditText.setText(selectedFeature.attributes["ST_STR_NAM"].toString())
+      setPositiveButton("OK") { _, _ ->
+        // set AD_ADDRESS value to the int from the edit text
+        selectedFeature.attributes["AD_ADDRESS"] =
+          editAttributeView.addressNumberEditText.text.toString().toInt()
```
Comment (hud10837) [inline kotlin/edit-features-with-feature-linked-annotation/src/main/java/com/esri/arcgisruntime/sample/editfeatureswithfeaturelinkedannotation/MainActivity.kt:176]: I think you're 100% right but I just can't believe this is how you get an int from an edittext

File: kotlin/edit-features-with-feature-linked-annotation/src/main/java/com/esri/arcgisruntime/sample/editfeatureswithfeaturelinkedannotation/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+        // set ST_STR_NAM value to the string from edit text
+        selectedFeature.attributes["ST_STR_NAM"] = editAttributeView.streetEditText.text.toString()
+        selectedFeature.featureTable?.updateFeatureAsync(selectedFeature)
```
Comment (hud10837) [inline kotlin/edit-features-with-feature-linked-annotation/src/main/java/com/esri/arcgisruntime/sample/editfeatureswithfeaturelinkedannotation/MainActivity.kt:179]: This line could use a comment, just so it's highlighted mainly

File: kotlin/edit-features-with-feature-linked-annotation/src/main/java/com/esri/arcgisruntime/sample/editfeatureswithfeaturelinkedannotation/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+   * Move the last of the vertex point of the currently selected polyline to the given map point, by updating the selected
+   * feature's geometry and feature table.
+   *
+   * @param mapPoint to which the last point of the polyline should be moved to
+   */
+  private fun movePolylineVertex(mapPoint: Point) {
+    val polyline = selectedFeature?.geometry as Polyline
+    val polylineBuilder = PolylineBuilder(polyline)
+    // get the first part of the polyline
+    polylineBuilder.parts[0].apply {
```
Comment (hud10837) [inline kotlin/edit-features-with-feature-linked-annotation/src/main/java/com/esri/arcgisruntime/sample/editfeatureswithfeaturelinkedannotation/MainActivity.kt:211]: Addressed this in the design but I find this particular usage quite confusing. Is there any way to perhaps move the entire line to some position? That might be too complex

File: kotlin/edit-features-with-feature-linked-annotation/src/main/res/values/colors.xml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (hud10837) [inline kotlin/edit-features-with-feature-linked-annotation/src/main/res/values/colors.xml:1]: Looks like you forgot to update the colors

**Comment by TADraeseke** [comment]
Thanks @hud10837 - Made those fixes, including behavioural improvements:

- Disallows selection of multi segmented polylines (too confusing to detail behaviour for and out of the scope of this sample)
- The nearest vertex to the tapped point of a polyline will be the one moved on next tap

File: kotlin/edit-features-with-feature-linked-annotation/src/main/java/com/esri/arcgisruntime/sample/editfeatureswithfeaturelinkedannotation/MainActivity.kt [hunk 5]
```diff
@@ -136,7 +138,19 @@ class MainActivity : AppCompatActivity() {
         (result.layerContent as? FeatureLayer)?.let { featureLayer ->
           // get a reference to the identified feature
           selectedFeature = result.elements[0] as? Feature
-          selectedFeature?.also {
+          // if the selected feature is a polyline with more than one segment
+          if ((selectedFeature?.geometry is Polyline)
+            && (selectedFeature?.geometry as Polyline).parts[0].pointCount > 2
```
Comment (hud10837) [inline kotlin/edit-features-with-feature-linked-annotation/src/main/java/com/esri/arcgisruntime/sample/editfeatureswithfeaturelinkedannotation/MainActivity.kt:143]: This seems like a fine solution and I think I checked it with every curved polyline in the sample, but is it technically possible for a curved polyline to have exactly two points in its first part but more points in other parts?

File: kotlin/edit-features-with-feature-linked-annotation/src/main/java/com/esri/arcgisruntime/sample/editfeatureswithfeaturelinkedannotation/MainActivity.kt [hunk 6]
```diff
@@ -205,12 +220,18 @@ class MainActivity : AppCompatActivity() {
...
   private fun movePolylineVertex(mapPoint: Point) {
+    // get the selected feature's geometry as a polyline
     val polyline = selectedFeature?.geometry as Polyline
+    // get the nearest vertex to the map point on the selected feature polyline
+    val nearestVertex = GeometryEngine.nearestVertex(
+      polyline,
+      GeometryEngine.project(mapPoint, polyline.spatialReference) as Point
+    )
     val polylineBuilder = PolylineBuilder(polyline)
     // get the first part of the polyline
```
Comment (hud10837) [inline kotlin/edit-features-with-feature-linked-annotation/src/main/java/com/esri/arcgisruntime/sample/editfeatureswithfeaturelinkedannotation/MainActivity.kt:231]:
```suggestion
    // get the part of the polyline associated with that vertex
```

File: kotlin/edit-features-with-feature-linked-annotation/src/main/java/com/esri/arcgisruntime/sample/editfeatureswithfeaturelinkedannotation/MainActivity.kt (hunk 5 above)
Comment (TADraeseke) [reply kotlin/edit-features-with-feature-linked-annotation/src/main/java/com/esri/arcgisruntime/sample/editfeatureswithfeaturelinkedannotation/MainActivity.kt:143]: Hmm...that's a very good point...

**Comment by TADraeseke** [comment]
Thanks @hud10837 - Really good point on first polyline part segments! I added a loop to check for curves on all parts instead. Thanks for spotting it! Merging now!

**Pull request merged**